MantidQt/MantidWidgets/src/MuonFitPropertyBrowser.cpp

```diff
@@ -380,6 +380,27 @@ void MuonFitPropertyBrowser::enumChanged(QtProperty *prop) {
     int j = m_enumManager->value(m_workspace);
     std::string option = m_workspaceNames[j].toStdString();
     setOutputName(option);
+    if (m_periodBoxes.size() > 1) {
+      size_t end = 0;
+      // assumed structure of name
+      // isolate the period
+      for (int k = 0; k < 4; k++) {
+        end = option.find_first_of(";");
+        option = option.substr(end + 1, option.size());
+      }
+      end = option.find_first_of(";");
+      QString selectedPeriod = QString::fromStdString(option.substr(0, end));
+      // turn on only the relevant box
+      for (auto iter = m_periodBoxes.constBegin();
+           iter != m_periodBoxes.constEnd(); ++iter) {
+        if (selectedPeriod == iter.key()) {
+          m_boolManager->setValue(iter.value(), true);
+        } else {
+          m_boolManager->setValue(iter.value(), false);
```

Comment:
I don't think the conditional is necessary, it should compile just fine without it.